sdk/src/types.rs

```diff
@@ -27,6 +27,28 @@ use std::{
     time::{Duration, SystemTime, UNIX_EPOCH},
 };
 
+#[derive(Debug)]
+enum LocalAccountAuthenticator {
+    PrivateKey(AccountKey),
+    // TODO: Add support for keyless authentication
+}
+
+impl LocalAccountAuthenticator {
+    pub fn sign_transaction(&self, txn: RawTransaction) -> SignedTransaction {
```

Comment:
do you know if AccountAuthenticator will support keyless?
basically can we change functions in RawTransaction to accept AccountAuthenticator’s instead of (address, private keys)?

and then instead of LocalAccountAuthenticator providing sign_transaction method, it can provide get_authenticator() -> AccountAuthenticator method instead, which will make all sign_transaction/sign_multi_agent/sign_fee_payer work

Reply:
As discussed, we can simplify this once we add keyless support.